src/pb_stub.cc

```diff
@@ -1101,13 +1106,76 @@ PYBIND11_EMBEDDED_MODULE(c_python_backend_utils, module)
             py::object loop =
                 py::module_::import("asyncio").attr("get_running_loop")();
             py::cpp_function callback = [infer_request]() {
-              auto response = infer_request->Exec();
-              return response;
+              auto responses = infer_request->Exec(false /* is_stream*/);
+              return responses[0];
             };
             py::object future =
                 loop.attr("run_in_executor")(py::none(), callback);
             return future;
           })
+      .def(
+          "stream_exec",
+          [](std::shared_ptr<InferRequest>& infer_request,
+             const int32_t execution_timeout) {
```

Comment:
On a second thought, for the execution timeout, it is possible that the backend side cannot properly clean up the objects if the process is terminated when exceeding the execution timeout. I was wondering if the execution timeout should be added to the API or if there is some WAR for this.

Reply:
I think for timeout you can use the server API to set the timeout: https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonserver.h#L1088 If the request times out, the request will respond with an error.

Reply:
Great suggestion! Updated.